backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -65,17 +66,24 @@
      */
     private final Stack<SessionFrame> frameStack;
 
+    private Hashtable<String, String> customSessionData;
+
     public CommCareSession(CommCarePlatform platform) {
         this.platform = platform;
         collectedDatums = new OrderedHashtable();
         this.frame = new SessionFrame();
         this.frameStack = new Stack<SessionFrame>();
+        this.customSessionData = new Hashtable<>();
     }
 
     public Vector<Entry> getEntriesForCommand(String commandId) {
         return this.getEntriesForCommand(commandId, new OrderedHashtable());
     }
 
+    public void putSessionDatum(String key, String value){
```

Comment:
Call this anything but 'datum', there are already too many datas and datums in a session. Try like putCustomSessionValue() or something like that.